be1-go/hub/standard_hub/message_handling.go

```diff
@@ -189,21 +189,16 @@ func (h *Hub) handleAnswer(senderSocket socket.Socket, byteMessage []byte) error
 		return nil
 	}
 
-	h.queries.Lock()
-
-	val := h.queries.state[*answerMsg.ID]
+	val := h.queries.GetQueryState(*answerMsg.ID)
 	if val == nil {
-		h.queries.Unlock()
 		return xerrors.Errorf("no query sent with id %v", answerMsg.ID)
 	}
 
 	if *val {
-		h.queries.Unlock()
 		return xerrors.Errorf("query %v already got an answer", answerMsg.ID)
 	}
 
-	*h.queries.state[*answerMsg.ID] = true
-	h.queries.Unlock()
+	h.queries.SetQueryState(*answerMsg.ID, true)
```

Comment:
I agree it looks terrible. It might've made sense somehow when the code was first introduced (a few semesters ago) but I can't find the reason. Especially since I added Set and Get functions now it's definitely better if I also move the logic inside.